src/NCronJob/Configuration/Builder/DependencyBuilder.cs

```diff
@@ -20,10 +20,43 @@ public sealed class DependencyBuilder<TPrincipalJob>
     public DependencyBuilder<TPrincipalJob> RunJob<TJob>(object? parameter = null)
         where TJob : IJob
     {
-        dependentJobOptions.Add(new JobDefinition(typeof(TJob), parameter, null, null));
+        var jobDefinition = jobRegistry.FindJobDefinition(typeof(TJob));
+
+        if (jobDefinition is null)
+        {
+            dependentJobOptions.Add(new JobDefinition(typeof(TJob), parameter, null, null));
+        }
+        else
+        {
+            dependentJobOptions.Add(jobDefinition with { Parameter = parameter });
+        }
+
         return this;
     }
 
+    /// <summary>
+    /// Adds a job that runs after the principal job has finished with a given <paramref name="parameter"/>.
+    /// <param name="jobName">The name of the registered job to run.</param>
+    /// <param name="parameter">An optional parameter that is passed down as the <see cref="JobExecutionContext"/> to the job.</param>
+    /// </summary>
+    /// <remarks>
+    /// </remarks>
+    public DependencyBuilder<TPrincipalJob> RunJob(string jobName, object? parameter = null)
```

Comment:
I see where you are driving at. Imagine the following registration:
```csharp
builder.Services.AddNCronJob(o => 
{
  // Here comes your new function into play
  o.AddJob<...>(w => w.WithCronExpression("* * * * *")).ExecuteWhen(s => s.RunJob("Name", null));

  o.AddJob<DependentJob>(o => o.WithCronExpression("* * * * *").WithName("Name"));
});
```

**Technically** the user registered a job with the given job name. The order inside a container registration shouldn't be important. 

And that is where your `useNCronJobAsync` proposal from a few days may come in handy.

Reply:
But this is more of an exotic use case (at least in my mind).

Reply:
>But this is more of an exotic use case (at least in my mind).

I'm not sure. I've tried a spike where all dependencies were required to be registered before jobs leveraging them.
Although that works, that's an awful user experience.

> The order inside a container registration shouldn't be important.

I have an idea how to resolve this